### PR TITLE
Prometheus events #506

### PR DESCRIPTION
**Pull request by Buzzglo**

### CODE REVIEW

File: src/gluon/events/project/ProjectCreated.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
 import {
+    addressSlackChannelsFromContext,
+    buttonForCommand,
     EventFired,
     HandlerContext,
     HandlerResult,
     logger,
 } from "@atomist/automation-client";
+
```
Comment (andrejonathan) [inline src/gluon/events/project/ProjectCreated.ts]: Should we have an empty line here?

**Comment by andrejonathan** [comment]
@Buzzglo Tested core features as well as Add/Remove user and seems good

File: src/gluon/events/member/TeamMemberCreated.ts [hunk 2]
```diff
@@ -24,11 +25,16 @@ subscription TeamMemberCreatedEvent {
...
     public async handle(event: EventFired<any>, ctx: HandlerContext): Promise<HandlerResult> {
         logger.info(`Ingested TeamMemberCreated event: ${JSON.stringify(event.data)}`);
-
-        return await SuccessPromise;
+        try {
+            this.succeedEvent();
+            return await SuccessPromise;
+        } catch (error) {
+            this.failEvent();
+            throw new Error(error);
```
Comment (kbristow) [inline src/gluon/events/member/TeamMemberCreated.ts]: Just rethrow the error here
```suggestion
            throw error;
```

File: src/gluon/events/project/ProjectProductionEnvironmentsRequested.ts [hunk 3]
```diff
@@ -56,7 +55,7 @@ export class ProjectProductionEnvironmentsRequested implements HandleEvent<any>
 
             for (const teamMember of membersToMessage) {
                 const requestCorrelationId: string = uuid();
-                const destination = await addressSlackUsersFromContext(ctx, teamMember.slack.screenName);
+                const destination =  await addressSlackUsersFromContext(ctx, teamMember.slack.screenName);
```
Comment (kbristow) [inline src/gluon/events/project/ProjectProductionEnvironmentsRequested.ts]: Remove the extra space
```suggestion
                const destination = await addressSlackUsersFromContext(ctx, teamMember.slack.screenName);
```

File: src/gluon/events/project/TeamAssociated.ts [hunk 4]
```diff
@@ -31,16 +33,21 @@ subscription TeamsLinkedToProjectEvent {
...
         logger.info(`Ingested TeamAssociated event: ${JSON.stringify(event.data)}`);
 
-        const teamsLinkedToProjectEvent = event.data.TeamsLinkedToProjectEvent[0];
+        try {
+            const teamsLinkedToProjectEvent = event.data.TeamsLinkedToProjectEvent[0];
 
-        const destination = await addressSlackChannelsFromContext(ctx, teamsLinkedToProjectEvent.team[0].slackIdentity.teamChannel);
-        return ctx.messageClient.send(`Your team has been successfully associated with ${teamsLinkedToProjectEvent.id}`,
-            destination);
+            const destination =  await addressSlackChannelsFromContext(ctx, teamsLinkedToProjectEvent.team[0].slackIdentity.teamChannel);
```
Comment (kbristow) [inline src/gluon/events/project/TeamAssociated.ts]: Another extra space for some reason.
```suggestion
            const destination = await addressSlackChannelsFromContext(ctx, teamsLinkedToProjectEvent.team[0].slackIdentity.teamChannel);
```

File: src/gluon/events/project/ProjectCreated.ts [hunk 5]
```diff
@@ -45,50 +45,55 @@ subscription ProjectCreatedEvent {
...
-        associateTeamCommand.projectName = projectCreatedEvent.project.name;
+            const associateTeamCommand = new AssociateTeam();
+            associateTeamCommand.projectName = projectCreatedEvent.project.name;
 
-        const destination = await addressSlackChannelsFromContext(ctx, projectCreatedEvent.team.slackIdentity.teamChannel);
-        return await ctx.messageClient.send({
-            text: `The *${projectCreatedEvent.project.name}* project has been created successfully.`,
-            attachments: [{
-                text: `
+            const destination =  await addressSlackChannelsFromContext(ctx, projectCreatedEvent.team.slackIdentity.teamChannel);
```
Comment (kbristow) [inline src/gluon/events/project/ProjectCreated.ts]:
```suggestion
            const destination = await addressSlackChannelsFromContext(ctx, projectCreatedEvent.team.slackIdentity.teamChannel);
```

File: src/gluon/events/project/ProjectProductionEnvironmentsRequestClosed.ts [hunk 6]
```diff
@@ -91,7 +90,8 @@ export class ProjectProductionEnvironmentsRequestClosed implements HandleEvent<a
             await handleQMError(qmMessageClient, error);
             const project = await this.gluonService.projects.gluonProjectFromProjectName(projectProdRequest.project.name);
             const correlationId: string = uuid();
-            const destination = await addressSlackChannelsFromContext(ctx, project.owningTeam.slack.teamChannel);
+            const destination =  await addressSlackChannelsFromContext(ctx, project.owningTeam.slack.teamChannel);
```
Comment (kbristow) [inline src/gluon/events/project/ProjectProductionEnvironmentsRequestClosed.ts]:
```suggestion
            const destination = await addressSlackChannelsFromContext(ctx, project.owningTeam.slack.teamChannel);
```

File: src/gluon/events/team/MembershipRequestCreated.ts [hunk 7]
```diff
@@ -85,10 +86,12 @@ export class MembershipRequestCreated implements HandleEvent<any> {
                 }],
             };
             logger.info(membershipRequestCreatedEvent.team.slackIdentity.teamChannel);
-            const destination = await addressSlackChannelsFromContext(ctx, membershipRequestCreatedEvent.team.slackIdentity.teamChannel);
+            const destination =  await addressSlackChannelsFromContext(ctx, membershipRequestCreatedEvent.team.slackIdentity.teamChannel);
```
Comment (kbristow) [inline src/gluon/events/team/MembershipRequestCreated.ts]:
```suggestion
            const destination = await addressSlackChannelsFromContext(ctx, membershipRequestCreatedEvent.team.slackIdentity.teamChannel);
```

File: src/gluon/metrics/prometheus/PrometheusClient.ts [hunk 8]
```diff
@@ -43,10 +43,10 @@ export class PrometheusClient {
...
 
                 const eventCounter = new Counter({
-                    name: `${eName}_event`,
-                    help: `Count the number of times the ${event.name} event fires`,
-                    labelNames: ["classname"],
-                });
+                name: `${eName}_event`,
+                help: `Count the number of times the ${event.name} event fires`,
+                labelNames: ["classname", "status"],
+            });
```
Comment (kbristow) [inline src/gluon/metrics/prometheus/PrometheusClient.ts]: Indentation looks odd here. Could you just double check formatting is correct?

**Pull request merged**